File: src/main/java/ua/softgroup/matrix/server/supervisor/producer/json/v2/ProjectWorkingDay.java
package ua.softgroup.matrix.server.supervisor.producer.json.v2;

import java.time.LocalDate;
import java.util.Set;

/**
 * @author Oleksandr Tyshkovets <sg.olexander@gmail.com>
 */
public class ProjectWorkingDay {

    private LocalDate date;
    private int totalDayWorkTimeSeconds;
    private int totalIdleTimeSeconds;
    private double totalIdlePercentage;
    private Set<Executor> executors;

    public LocalDate getDate() {
        return date;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    public int getTotalDayWorkTimeSeconds() {
        return totalDayWorkTimeSeconds;
    }

    public void setTotalDayWorkTimeSeconds(int totalDayWorkTimeSeconds) {
        this.totalDayWorkTimeSeconds = totalDayWorkTimeSeconds;
    }

    public int getTotalIdleTimeSeconds() {
        return totalIdleTimeSeconds;
    }

    public void setTotalIdleTimeSeconds(int totalIdleTimeSeconds) {
        this.totalIdleTimeSeconds = totalIdleTimeSeconds;
    }

    public double getTotalIdlePercentage() {
        return totalIdlePercentage;
    }

    public void setTotalIdlePercentage(double totalIdlePercentage) {
        this.totalIdlePercentage = totalIdlePercentage;
    }

    public Set<Executor> getExecutors() {
        return executors;
    }

    public void setExecutors(Set<Executor> executors) {
        this.executors = executors;
    }
}

File: src/main/scala/ua/softgroup/matrix/server/supervisor/producer/json/v2/ProjectWorkingDay.scala
package ua.softgroup.matrix.server.supervisor.producer.json.v2

import java.time.LocalDate
import java.util

import scala.beans.BeanProperty

/**
  * @author Oleksandr Tyshkovets <olexandr.tyshkovets@gmail.com>
  */
class ProjectWorkingDay {

  @BeanProperty
  var date: LocalDate = _

  @BeanProperty
  var totalDayWorkTimeSeconds: Int = _

  @BeanProperty
  var totalIdleTimeSeconds: Int = _

  @BeanProperty
  var totalIdlePercentage: Double = _

  @BeanProperty
  var executors: util.Set[Executor] = _

  override def toString: String = {
    "ProjectWorkingDay(date=%s, totalDayWorkTimeSeconds=%d, totalIdleTimeSeconds=%d, totalIdlePercentage=%f, executors=%s)"
      .format(date, totalDayWorkTimeSeconds, totalIdleTimeSeconds, totalIdlePercentage, executors)
  }

}
